### DISCUSSION TITLE
Ability to wipe the environment periodically

### DISCUSSION THREAD

**Discussion by davidbarton** [pockethost/pockethost#393] [Q&A] [answered]
I have created a demo to showcase cookie based authentication flow in Pocketbase https://github.com/davidbarton/pocketbase-cookie-auth-demo.

In order to reduce potential abuse I would like to wipe the environment clean in fixed intervals and start with a fresh state, similar to other playground like projects. The method to achieve it does not really matter. It can be an admin UI config, or cron terminal command, or Github action re-deploy, all are great. Yet I was not able to come up with any way to achieve that.

@benallfree Is something like this possible today or in future?

**Comment by benallfree** [comment]
I'd suggest maybe a `pb_hook` for this purpose. Are you looking to reset the db to a known state?

Alternatively, if the instance is in maintenance mode, you can always delete data.db via FTP. At next start, all the migrations will run again.

**Reply by davidbarton** [reply, marked as answer]
With some custom code, `pb_hooks` would allow to reset the db to a known state as you suggest. It is a good enough option and I will likely end up with this one.

Yet it is not what my goal was. My goal was to completely wipe all state from the environment and start from zero. Same as creating brand new project on PocketHost. That way I would be sure to eliminate all possible abuse since nothing will persist.

Using the FTP (or other means) to clear all files would likely work but that would require a programatic way to enable/disable the maintenance state.

Anyway unless you know about more destructive option now I will probably go with `pb_hooks` and known db state option for now.
